admin/create-theme/theme-fonts.php

```diff
@@ -3,6 +3,66 @@
 
 class CBT_Theme_Fonts {
 
+
+	/**
+	 * Make the font face theme src urls absolute.
+	 *
+	 * It replaces the 'file:./' prefix with the theme directory uri.
+	 *
+	 * Example: 'file:./assets/fonts/my-font.ttf' -> 'http://example.com/wp-content/themes/my-theme/assets/fonts/my-font.ttf'
+	 * Example: [ 'https://example.com/assets/fonts/my-font.ttf' ] -> [ 'https://example.com/assets/fonts/my-font.ttf' ]
+	 *
+	 * @param array|string $src
+	 * @return array|string
+	 */
+	public static function make_theme_font_src_absolute( $src ) {
+		$make_absolute = function ( $url ) {
+			if ( str_starts_with( $url, 'file:./' ) ) {
+				return str_replace( 'file:./', get_stylesheet_directory_uri() . '/', $url );
+			}
+			return $url;
+		};
+
+		if ( is_array( $src ) ) {
+			foreach ( $src as &$url ) {
+				$url = $make_absolute( $url );
+			}
+		} else {
+			$src = $make_absolute( $src );
+		}
+
+		return $src;
+	}
+
+	/**
+	 * Get all fonts from the theme.json data + all the style variations.
+	 *
+	 * @return array
+	 */
+	public static function get_all_fonts() {
+		$font_families = array();
+		$theme         = CBT_Theme_JSON_Resolver::get_merged_data();
+		$settings      = $theme->get_settings();
+
+		if ( isset( $settings['typography']['fontFamilies']['theme'] ) ) {
+			$font_families = array_merge( $font_families, $settings['typography']['fontFamilies']['theme'] );
+		}
+
+		if ( isset( $settings['typography']['fontFamilies']['custom'] ) ) {
+			$font_families = array_merge( $font_families, $settings['typography']['fontFamilies']['custom'] );
+		}
+
+		$variations = CBT_Theme_JSON_Resolver::get_style_variations();
+
+		foreach ( $variations as $variation ) {
+			if ( isset( $variation['settings']['typography']['fontFamilies']['theme'] ) ) {
+				$font_families = array_merge( $font_families, $variation['settings']['typography']['fontFamilies']['theme'] );
+			}
+		}
+
+		return $font_families;
```

Comment:
If we drop something like this into this `get_all_fonts` function then we can eliminate the 'make theme absolute' function (which is really just string/array checks and a `str_replace`), eliminate the iteration logic in the api class and always ensure that the src attributes are arrays.

```
		// ensure that src is an array
		foreach( $font_families as &$font_family ) {
			if ( isset( $font_family['fontFace'] ) ) {
				foreach( $font_family['fontFace'] as &$font_face ) {
					if ( ! is_array( $font_face['src'] ) ) {
						$font_face['src'] = array( $font_face['src'] );
					}
					// make the src urls absolute
					if ( $absolute_src ) {
						foreach( $font_face['src'] as &$font_face_src ) {
							if ( str_starts_with( $font_face_src, 'file:./' ) ) {
								return str_replace( 'file:./', get_stylesheet_directory_uri() . '/', $font_face_src );
							}
						}
					}
				}
			}
		}
```

Reply:
> If we drop something like this into this get_all_fonts function, we can eliminate the 'make theme absolute' .

I don't see the need to grow one function by absorbing the same code as another. The separation of concerns is clear. What's the benefit?

> ensure that the src attributes are arrays
.
Why we should ensure that?

Reply:
I see that you have already moved half of it in from the API class; that was most of the concern and that's enough that you can resolve this discussion if you want.

But I find the `make_theme_font_src_absolute` function overly complicated when the same could be completed using fewer, easier to understand lines in the `get_all_fonts` function.  The `make_theme_font_src_absolute` logic is not re-used.

By ensuring that the response to the API always returns the src in the form of an array the client won't have to execute the "is it a string or an array" check when it works with this data; it will always be an array.